Formula/apollo-cli.rb

```diff
@@ -0,0 +1,20 @@
+require "language/node"
+
+class ApolloCli < Formula
+  desc "Tooling for development and production Apollo workflows"
+  homepage "https://apollographql.com"
+  url "https://registry.npmjs.org/apollo/-/apollo-2.17.3.tgz"
+  sha256 "1fd8c82904e3b70a7133ce30f29545e707468809c6b2cef477836ef1afcd2d2c"
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/apollo --version")
+    assert_match /apollo/, shell_output("#{bin}/apollo help")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.